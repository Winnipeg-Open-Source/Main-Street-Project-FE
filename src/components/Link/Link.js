import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Link as RouterLink } from 'react-router-dom';
import { getPaletteColor, mediaQueries } from 'pcln-design-system';

const StyledLink = styled(RouterLink)`
    display: flex;
    align-items: center;
    color: ${(props) =>
        props.isActive
            ? getPaletteColor('text.darkest')
            : getPaletteColor('text.light')};
    font-size: 20px;
    font-weight: bold;

    & > svg {
        margin-right: 16px;
    }

    &:hover {
        color: ${getPaletteColor('text.darkest')};
    }

    ${mediaQueries.sm} {
        font-size: 16px;
        font-weight: normal;
        color: ${(props) =>
            props.isActive
                ? getPaletteColor('text.darkest')
                : getPaletteColor('text.lightest')};
    }
`;

function Link ({ className, isActive, to, icon, children, onClick }) {
    return (
        <StyledLink
            className={className}
            isActive={isActive}
            to={to}
            onClick={onClick}
        >
            {icon}
            {children}
        </StyledLink>
    );
}

Link.displayName = 'Link';

Link.propTypes = {
    className: PropTypes.string,
    isActive: PropTypes.bool,
    to: PropTypes.string.isRequired,
    icon: PropTypes.node,
    children: PropTypes.node,
    onClick: PropTypes.func,
};

Link.defaultProps = {
    className: '',
};

export default memo(Link);
