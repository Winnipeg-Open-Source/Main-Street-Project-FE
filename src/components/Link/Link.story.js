import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Home } from 'pcln-icons';
import Link from 'components/Link';

export const Default = () => (
    <BrowserRouter>
        <Link>Link Text</Link>
    </BrowserRouter>
);
export const WithIcon = () => (
    <BrowserRouter>
        <Link icon={<Home />}>Link Text</Link>
    </BrowserRouter>
);

export default {
    title: 'Link',
    component: Link,
};
