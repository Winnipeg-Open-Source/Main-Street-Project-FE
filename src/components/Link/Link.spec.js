import React from 'react';
import renderWithThemeAndRouter from 'tests/helpers/renderWithThemeAndRouter';
import { Home } from 'pcln-icons';
import Link from './Link';

describe('Link', () => {
    it('renders without crashing', () => {
        const { asFragment } = renderWithThemeAndRouter(<Link />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders without icon', () => {
        const { getByText, queryByTestId } = renderWithThemeAndRouter(
            <Link>Link Text</Link>
        );

        expect(getByText(/Link Text/i)).toBeInTheDocument();
        expect(queryByTestId(/icon/)).not.toBeInTheDocument();
    });

    it('renders with icon', () => {
        const { getByText, queryByTestId } = renderWithThemeAndRouter(
            <Link icon={<Home data-testid='icon' />}>Link Text</Link>
        );

        expect(getByText(/Link Text/i)).toBeInTheDocument();
        expect(queryByTestId(/icon/)).toBeInTheDocument();
    });
});
