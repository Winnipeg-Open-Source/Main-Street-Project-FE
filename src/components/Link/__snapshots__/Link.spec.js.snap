// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-qOvoE hIfWnJ sc-qWSGh dpgLb"
  >
    <a
      class="sc-pAMRk fwRIZg"
      href=""
    />
  </div>
</DocumentFragment>
`;
